src/io.rs

```diff
@@ -106,3 +109,78 @@ impl<D: Drive> Future for Print<D> {
         }
     }
 }
+
+/// A handle to the standard output of the current process.
+pub struct Stdout<D: Drive> {
+    ring: Ring<D>,
+    buf: Buffer,
+}
+
+/// Constructs a new handle to the standard output of the current process using the demo driver.
+/// ```no_run
+/// use ringbahn::io;
+///
+/// # use futures::AsyncWriteExt;
+/// # fn main() -> std::io::Result<()> { futures::executor::block_on(async {
+/// io::stdout().write(b"hello, world").await?;
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+// TODO synchronization note?
+pub fn stdout() -> Stdout<DemoDriver> {
+    Stdout::run_on_driver(DemoDriver::default())
+}
+
+impl<D: Drive> Stdout<D> {
+    pub fn run_on_driver(driver: D) -> Stdout<D> {
+        Stdout {
+            ring: Ring::new(driver),
+            buf: Buffer::default(),
+        }
+    }
+
+    #[inline(always)]
+    fn split(self: Pin<&mut Self>) -> (Pin<&mut Ring<D>>, &mut Buffer) {
+        unsafe {
+            let this = Pin::get_unchecked_mut(self);
+            (Pin::new_unchecked(&mut this.ring), &mut this.buf)
+        }
+    }
+}
+
+impl<D: Drive> AsyncWrite for Stdout<D> {
+    fn poll_write(self: Pin<&mut Self>, ctx: &mut Context<'_>, slice: &[u8])
+        -> Poll<io::Result<usize>>
+    {
+        let fd = self.as_raw_fd();
+        let (ring, buf, ..) = self.split();
+        let data = ready!(buf.fill_buf(|mut buf| {
+            Poll::Ready(Ok(io::Write::write(&mut buf, slice)? as u32))
+        }))?;
+        let n = ready!(ring.poll(ctx, 1, |sqs| {
+            let mut sqe = sqs.single().unwrap();
+            unsafe {
+                sqe.prep_write(fd, data, 0);
+            }
+            sqe
+        }))?;
+        buf.clear();
+        Poll::Ready(Ok(n as usize))
+    }
+
+    fn poll_flush(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
```

Comment:
Is this the right way to flush stdout? I'm not sure what the right semantics are.

Reply:
I am also not entirely sure if this is correct, do you mean that we might need to `drain` the io-uring? async-std's [implementation](https://github.com/async-rs/async-std/blob/master/src/io/stdout.rs#L141-L168) has a lot more going on but I was hoping all the state logic was taken care of by `Ring`.

Reply:
nevermind, I think! I thought that linebuffering of stdout was performed by the kernel, but after some googling I think its just performed by libc. I think this is correct